bayesian/sentiment_analysis.py

```diff
@@ -0,0 +1,141 @@
+        pass
+
+    def get_latest_comment_query(self, pkg_name):
+        latest_comment_query = "SELECT  text, creation_date FROM `bigquery-public-data.stackoverflow.comments` \
+        where text like '%" + pkg_name + "%' ORDER BY creation_date desc LIMIT 1"
+        return latest_comment_query
+
+    def get_comment_query(self, pkg_name):
+        comment_query = "SELECT text FROM `bigquery-public-data.stackoverflow.comments` \
+        where text like '%" + pkg_name + "%'"
+        return comment_query
+
+    def get_question_query(self, pkg_name):
+        question_query = "SELECT body FROM `bigquery-public-data.stackoverflow.posts_questions` \
+        where body like '%" + pkg_name + "%'"
+        return question_query
+
+    def get_answer_query(self, pkg_name):
+        answer_query = "SELECT body FROM `bigquery-public-data.stackoverflow.posts_answers` \
+        where body like '%" + pkg_name + "%'"
+        return answer_query
+
+    def get_stackoverflow_query(self, pkg_name):
+        stackoverflow_query = "SELECT body FROM `bigquery-public-data.stackoverflow.stackoverflow_posts` \
+        where body like '%" + pkg_name + "%'"
+        return stackoverflow_query
+
+    def query_big_query(self, key_file, bigquerysql):
+        """
+        Execute queries related to fetch StackOverflow data by using google big query
+        :param key_file:
+        :param bigquerysql:
+        :return: Query response
+        """
+        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = key_file
+        bigquery_service = bigquery.Client()
+        query = bigquery_service.run_sync_query(bigquerysql)
+        query.timeout_ms = 60000
+        query.use_legacy_sql = False
+        query.use_query_cache = True
+        query.run()
+        return  query
+
+    def get_sentiment_score(self, all_text):
+        yes_sentiment_score = False 
+        language_service = language.Client()
+        score = 0.0
+        magnitude = 0.0
+        allcomment = ''
+        commentdatetime = ''
+
+        try:
+             document = language_service.document_from_text(all_text)
+             annotations = document.annotate_text(include_sentiment=True, include_syntax=False, include_entities=False)
+             score, magnitude = self.get_scores(annotations)
+
+        except Exception as e:
+            print(e)
+
+        return score, magnitude, all_text
+
+    def get_scores(self, annotations):
+        score = annotations.sentiment.score
+        magnitude = annotations.sentiment.magnitude
+        return score, magnitude
+
+    def get_aggregated_string(self, query_data):
+        aggregated_text = ''
+        for row in query_data.rows:
+            aggregated_text = aggregated_text + row[0] + ' '
+
+        return  aggregated_text
+
+    def get_latest_comment_time(self, query_data):
+        latest_comment_time = ''
+        for row in query_data.rows:
+            latest_comment_time = row[1]
+        return latest_comment_time
```

Comment:
You don't need to go through whole list to get the last item, see
https://stackoverflow.com/questions/930397/getting-the-last-element-of-a-list-in-python

so something like the following would be much faster
`return query_data.rows[-1][1] if query_data.rows else ''`